### PR TITLE
Psl 111 tracking generic result click

### PR DESCRIPTION
**Pull request by deomsj**
Add generic result click to the client trackers (already exists in the back end behavioral API v2)
 - This event should be sent exclusively from other click events
 - This event should be fired whenever a user clicks on a detail page HREF
 - Add event to constructorio-client-javascriptwire in through autocomplete-ui
 - Event properties
    - item_id: required
    - item_name: optional
    - variation_id: optional
    - section: optional, defaults to 'products'

Screen Shot 2021-06-04 at 8 25 03 PM

Note: tests are not passing consistently for me, but this is also true for other methods... is this a known issue? Any idea what could be causing this?

### CODE REVIEW

File: spec/src/modules/tracker.js [hunk 1]
```diff
@@ -3667,6 +3667,344 @@ describe('ConstructorIO - Tracker', () => {
     });
   });
 
+
+  describe('trackGenericResultClick', () => {
+    // Note: `variation_id` parameter not being passed as none are defined
```
Comment (jjl014) [inline spec/src/modules/tracker.js:3672]: What does `parameter not being passed as none are defined` mean? 🤔

File: spec/src/modules/tracker.js [hunk 2]
```diff
@@ -3667,6 +3667,344 @@ describe('ConstructorIO - Tracker', () => {
...
+        // Response
+        expect(eventSpy).to.have.been.called;
+        expect(responseParams).to.have.property('method').to.equal('POST');
+        expect(responseParams).to.have.property('message');
+
+        done();
+      }, waitInterval);
+    });
+
+    it('Should respond with a valid response when required and optional parameters are provided', (done) => {
```
Comment (jjl014) [inline spec/src/modules/tracker.js:3864]: I think this test description needs an update.

File: spec/src/modules/tracker.js [hunk 3]
```diff
@@ -3667,6 +3667,344 @@ describe('ConstructorIO - Tracker', () => {
...
+        // Response
+        expect(eventSpy).to.have.been.called;
+        expect(responseParams).to.have.property('method').to.equal('POST');
+        expect(responseParams).to.have.property('message').to.equal('ok');
+
+        done();
+      }, waitInterval);
+    });
+
+    it('Should respond with a valid response when required and optional parameters are provided', (done) => {
```
Comment (jjl014) [inline spec/src/modules/tracker.js:3899]: Same as above, test description needs an update?

**Review by jjl014 — COMMENTED** [review]
Nice work, this looks great! Just left a couple comments on the tests. 👍

**Comment by sblaurock** [comment]
Don't check in changes to `./docs` please! These will get compiled with the `version` command.

File: spec/src/modules/tracker.js (hunk 1 above)
Comment (sblaurock) [reply spec/src/modules/tracker.js]: `variation_id` is a valid parameter that can be passed to this method, however none are currently defined in the catalog we use for testing, so we aren't ever passing it in testing

This comment definitely could be more clear - I'll update in this PR. Good callout @jjl014 !

File: spec/src/modules/tracker.js (hunk 2 above)
Comment (sblaurock) [reply spec/src/modules/tracker.js:3864]: I think this isn't the only place that this mistake occurs - it's my fault originally!

**Review by sblaurock — APPROVED** [review]
Looks great to me! Nice work @deomsj. I made a few changes to help with the comments Jimmy raised - should be good to go. I'll merge and get this shipped 🚀

File: spec/src/modules/tracker.js (hunk 2 above)
Comment (sblaurock) [reply spec/src/modules/tracker.js:3864]: I've updated these

**Comment by deomsj** [comment]
awesome, thanks for the reviews and help getting this one across the finish line @jjl014  and @sblaurock!!!

**Pull request merged**